src/java/us/kbase/templates/module_python_impl.vm.properties

```diff
@@ -79,109 +63,150 @@ class ${module_name}:
-                                                              'workspace_name': workspace_name,
-                                                              'assembly_name': fasta_file['assembly_name']
-                                                              })
+        # replace the contigs in the contigSet object in local memory
+        contigSet['contigs'] = good_contigs
+        
+        print('Filtered ContigSet to '+str(n_remaining)+' contigs out of '+str(n_total))
+        
 
+        # Step 4- Save the new ContigSet back to the Workspace
+        # When objects are saved, it is important to always set the Provenance of that object.  The basic
+        # provenance info is given to you as part of the context object.  You can add additional information
+        # to the provenance as necessary.  Here we keep a pointer to the input data object.
+        provenance = [{}]
+        if 'provenance' in ctx:
+            provenance = ctx['provenance']
+        # add additional info to provenance here, in this case the input data object reference
+        provenance[0]['input_ws_objects']=[workspace_name+'/'+contigset_id]
+
+        obj_info_list = None
+        try:
+	        obj_info_list = wsClient.save_objects({
+	                            'workspace':workspace_name,
+	                            'objects': [
+	                                {
+	                                    'type':'KBaseGenomes.ContigSet',
+	                                    'data':contigSet,
+	                                    'name':contigset_id,
+	                                    'provenance':provenance
+	                                }
+	                            ]
+	                        })
+        except:
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            lines = traceback.format_exception(exc_type, exc_value, exc_traceback)
+            orig_error = ''.join('    ' + line for line in lines)
+            raise ValueError('Error saving filtered ContigSet object to workspace:\n' + orig_error)
+        
+        info = obj_info_list[0]
+        # Workspace Object Info is a tuple defined as-
+        # absolute ref = info[6] + '/' + info[0] + '/' + info[4]
+        # 0 - obj_id objid - integer valued ID of the object
+        # 1 - obj_name name - the name of the data object
+        # 2 - type_string type - the full type of the data object as: [ModuleName].[Type]-v[major_ver].[minor_ver]
+        # 3 - timestamp save_date
+        # 4 - int version - the object version number
+        # 5 - username saved_by
+        # 6 - ws_id wsid - the unique integer valued ID of the workspace containing this object
+        # 7 - ws_name workspace - the workspace name
+        # 8 - string chsum - md5 of the sorted json content
+        # 9 - int size - size of the json content
+        # 10 - usermeta meta - dictionary of string keys/values of user set or auto generated metadata
```

Comment:
it's confusing that you're talking about python and SDK types here, and the names of the fields are similar to the SDK types (`obj_id` vs `objid`). I find it a PITA to have to look up what actual type the data is (especially for numeric fields -- is it an int or is it a string containing an int?), and I think it would be more user-friendly just to put the python type of each field.